dodo.py

```diff
@@ -138,6 +139,100 @@ def task_noroot():
         ],
     }
 
+def task_venv():
+    '''
+    setup virtual env
+    '''
+    venv = f'{CFG.APP_PROJPATH}/.venv'
+    pip3 = f'{venv}/bin/pip3'
+    appreqs = f'{CFG.APP_PROJPATH}/requirements.txt'
+    testreqs = f'{CFG.APP_PROJPATH}/tests/requirements.txt'
+    return {
+        'task_dep': [
+            'noroot',
+        ],
+        'actions': [
+            f'virtualenv --python=$(which python3.7) {venv}',
+            f'{pip3} install --upgrade pip',
+            f'[ -f "{appreqs}" ] && {pip3} install -r "{appreqs}" || true',
+            f'[ -f "{testreqs}" ] && {pip3} install -r "{testreqs}" || true',
+        ]
+    }
+
+def task_dynalite():
+    '''
+    start, stop the dynalite db
+    '''
+    cmd = f'{DYNALITE} --port {CFG.DYNALITE_PORT}'
+    msg = f'dyanlite server started on {CFG.DYNALITE_PORT} logging to {CFG.DYNALITE_FILE}'
+    def running():
+        '''
+        return pid of running dyanlite
+        '''
+        pid = None
+        try:
+            pid = call(f'lsof -i:{CFG.DYNALITE_PORT} -t')[1].strip()
+        except CalledProcessError:
+            return None
+        try:
+            args = call(f'ps -p {pid} -o args=')[1].strip()
+        except CalledProcessError:
+            return None
+        if f'{cmd}' in args:
+            return pid
+        return None
+    pid = running()
+    yield {
+        'name': 'stop',
+        'task_dep': [
+            'noroot',
+            'check',
+            'npm',
+        ],
+        'actions': [
+            f'kill {pid}',
+        ],
+        'uptodate': [
+            lambda: pid is None
+        ],
+    }
+    yield {
+        'name': 'start',
+        'task_dep': [
+            'noroot',
+            'check',
+            'npm',
+        ],
+        'actions': [
+            LongRunning(f'nohup {cmd} > {CFG.DYNALITE_FILE} &'),
+            f'echo "{msg}"',
+        ],
+        'uptodate': [
+            lambda: pid,
+        ],
+    }
+
+def task_local():
+    '''
+    run locally
+    '''
+    python3 = f'{CFG.APP_PROJPATH}/.venv/bin/python3.7'
+    aws = ' '.join([
+        'AWS_ACCESS_KEY_ID=fake-id',
+        'AWS_SECRET_ACCESS_KEY=fake-key'
+    ])
+    return {
+        'task_dep': [
+            'noroot',
+            'venv',
+            'dynalite:start',
+        ],
+        'actions': [
+            f'{python3} -m setup develop',
+            f'{aws} {python3} subhub/app.py',
```

Comment:
This might be an Ubuntu on WSL thing, but to get this working I also had to tweak this like so:
```python
f'{aws} {python3} -m subhub.app',
```